src/backend/src/services/tasks.services.ts

```diff
@@ -258,4 +265,69 @@ export default class TasksService {
 
     return deletedTask.taskId;
   }
+
+  /**
+   * Sends the task deadline slack notifications for all tasks with a deadline of the given date
+   * @param deadline the deadline date as a number
+   */
+  static async sendTaskDeadlineSlackNotifications(deadline: number) {
+    const startOfDay = new Date(deadline);
+    const endOfDay = new Date(startOfDay);
+    endOfDay.setDate(startOfDay.getDate() + 1);
+
+    const tasks = await prisma.task.findMany({
+      where: {
+        deadline: {
+          gte: startOfDay,
+          lt: endOfDay
+        },
+        status: {
+          not: 'DONE'
+        }
+      },
+      include: {
+        assignees: {
+          include: {
+            userSecureSettings: true,
```

Comment:
why do you need the secure settings?

Reply:
typescript throws a type mismatch error without it since it is required in type UserWithSettings

Reply:
ohhh, UserWithSettings is a bad name cause it has secure setting and settings, you should change that to be named UserWithSecureSettings, and then make one with just settings that's name UserWithSettings, will probably be a bit annoying

Reply:
for reference:
<img width="375" alt="image" src="https://github.com/Northeastern-Electric-Racing/FinishLine/assets/29521172/edcd671b-6f8e-4f5b-b84e-bf21a5e83299">